Detectors/TPC/reconstruction/src/KrBoxClusterFinder.cxx

```diff
@@ -112,7 +86,53 @@ void KrBoxClusterFinder::fillADCValue(int cru, int rowInSector, int padInRow, in
     adcValue /= correctionFactor;
   }
 
-  mMapOfAllDigits[timeBin][rowInSector][corPad] = adcValue;
+  mTempTimeSlice[rowInSector][corPad] = adcValue;
+}
+
+void KrBoxClusterFinder::fillTempTimeSlice(std::vector<o2::tpc::Digit>& eventSector, const int timeSlice) {
+  // No third argument needed since iterator gets popped
+  for (std::vector<o2::tpc::Digit>::iterator digit = eventSector.begin(); digit != eventSector.end();) {
+    const int time = digit->getTimeStamp();
+    if(time != timeSlice) {
+      return;
+    }
+
+    const int cru = digit->getCRU();
+    mSector = cru / CRU::CRUperSector;
+
+    const int rowInSector = digit->getRow();
+    const int padInRow    = digit->getPad();
+    float adcValue        = digit->getChargeFloat();
+
+    fillADCValueInTempSlice(cru, rowInSector, padInRow, adcValue);
+
+    // We don't need this digit anymore:
+    digit = eventSector.erase(digit);
+  }
+  return;
+}
+
+const std::vector<KrCluster>& KrBoxClusterFinder::loopOverSector(std::vector<o2::tpc::Digit>& eventSector) {
+  mClusters.clear();
+  createInitialMap(eventSector);
+  for(int iTimeSlice = mMaxClusterSizeTime; iTimeSlice < MaxTimes - mMaxClusterSizeTime; ++iTimeSlice) {
+    findLocalMaxima(true, iTimeSlice);
+    popFirstTimeSliceFromMap();
+    fillTempTimeSlice(eventSector, iTimeSlice);
```

Comment:
You are right, yes!
This bug, however, had no big impact for the cluster finding, since it only affected the first few timebins. Nevertheless, I fixed it now. Thanks!